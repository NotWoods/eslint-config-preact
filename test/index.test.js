import { ESLint } from "eslint";
import * as fs from "node:fs";
import * as path from "node:path";
import { describe, test, expect } from "vitest";

const fixturesFolder = path.resolve(import.meta.dirname, "fixtures");
const cli = new ESLint({
  cwd: fixturesFolder,
  fix: false,
  cache: false,
  errorOnUnmatchedPattern: true,
});

const FIXTURES = fs
  .readdirSync(fixturesFolder, { withFileTypes: true })
  .filter((d) => d.isDirectory());

describe.each(FIXTURES)(`Fixture: $name`, (d) => {
  /** @param {string} f */
  const p = (f) => path.resolve(d.parentPath, d.name, f);

  test("valid", async () => {
    const lintResults = await cli.lintFiles(p("valid.js"));
    expect(lintResults).toHaveLength(1);
    expect(lintResults[0]).toHaveProperty("errorCount", 0);
    expect(lintResults[0]).toHaveProperty("warningCount", 0);

    const { filePath: _, ...snapshot } = lintResults[0];
    expect(snapshot).toMatchSnapshot();
  });

  test("invalid", async () => {
    const lintResults = await cli.lintFiles(p("invalid.js"));
    expect(lintResults).toHaveLength(1);
    expect(
      lintResults[0].errorCount + lintResults[0].warningCount,
    ).toBeGreaterThan(0);

    const { filePath: _, ...snapshot } = lintResults[0];
    expect(snapshot).toMatchSnapshot();
  });
});
