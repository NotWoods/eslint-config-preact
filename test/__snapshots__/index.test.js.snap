// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Fixture: 'component' > invalid 1`] = `
{
  "errorCount": 1,
  "fatalErrorCount": 0,
  "fixableErrorCount": 0,
  "fixableWarningCount": 0,
  "messages": [
    {
      "column": 5,
      "endColumn": 18,
      "endLine": 9,
      "line": 9,
      "message": "Do not use setState in componentDidMount",
      "messageId": "noSetState",
      "nodeType": "MemberExpression",
      "ruleId": "react/no-did-mount-set-state",
      "severity": 2,
    },
  ],
  "source": "import { Component } from "preact";

export class A extends Component {
  constructor() {
    super();
  }

  componentDidMount() {
    this.setState({});
  }
}
",
  "suppressedMessages": [],
  "usedDeprecatedRules": [],
  "warningCount": 0,
}
`;

exports[`Fixture: 'component' > valid 1`] = `
{
  "errorCount": 0,
  "fatalErrorCount": 0,
  "fixableErrorCount": 0,
  "fixableWarningCount": 0,
  "messages": [],
  "suppressedMessages": [],
  "usedDeprecatedRules": [],
  "warningCount": 0,
}
`;
